code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -418,34 +424,46 @@ func (mrh *MinerRestHandler) testNodeStat(w http.ResponseWriter, r *http.Request
 		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
 		return
 	}
-	var err error
-	var miner event.Miner
+
 	var dps []event.DelegatePool
-	if includeDelegates {
-		miner, dps, err = edb.GetMinerWithDelegatePools(id)
-	} else {
-		miner, err = edb.GetMiner(id)
-	}
-	if err == nil {
-		common.Respond(w, r, nodeStat{
-			NodeResponse: minerTableToMinerNode(miner, dps),
-			TotalReward:  int64(miner.Rewards.TotalRewards),
-		}, nil)
+
+	switch spenum.Provider(providerType) {
+	case spenum.Miner:
+		var miner event.Miner
+
+		if includeDelegates {
+			miner, dps, err = edb.GetMinerWithDelegatePools(provider_id)
+		} else {
+			miner, err = edb.GetMiner(provider_id)
+		}
+		if err == nil {
+			common.Respond(w, r, nodeStat{
+				NodeResponse: minerTableToMinerNode(miner, dps),
+				TotalReward:  int64(miner.Rewards.TotalRewards),
+			}, nil)
+		} else {
+			common.Respond(w, r, nil, common.NewErrBadRequest(err.Error()))
```

Comment:
In Go, we prefer to do err checking first and return error early rather than the `if else` format, so let's do
```go
if err != nil {
    common.Response(w, r, nil, common.NewErrBadRequest(err.Error)))
    return
}

common.Response(...)
return
```

Same for the sharder process below